src/scalar/utf16_to_latin1/valid_utf16_to_latin1.h

```diff
@@ -15,8 +15,13 @@ inline size_t convert_valid(const char16_t* buf, size_t len, char* latin_output)
 
   while (pos < len) {
     word = !match_system(big_endian) ? utf16::swap_bytes(data[pos]) : data[pos];
-    *latin_output++ = char(word);
-    pos++;
+    if (word <= 0xFF) {
+        *latin_output++ = char(word);
+        pos++;
+    } else {
+        // can't be represented as latin1 - return error
+        return 0;
```

Comment:
Well, now you are turning a function that assumes that the input is valid latin1 into one that validates.

Why would you use that function then when you have `convert_utf16[be/le]_to_latin1`?